modules/battery_manager/jsx/addEntry.js

```diff
@@ -0,0 +1,368 @@
+            />
+            <SearchableDropdown
+              name="forSite"
+              label="Site"
+              placeHolder="Search for site"
+              options={this.state.Data.sites}
+              strictSearch={true}
+              onUserInput={this.setFormData}
+              ref="forSite"
+              required={false}
+              value={this.state.formData.forSite}
+            />
+            <SelectElement
+              name="firstVisit"
+              label="First Visit"
+              options={this.state.Data.firstVisits}
+              onUserInput={this.setFormData}
+              ref="firstVisit"
+              required={false}
+              value={this.state.formData.firstVisit}
+            />
+            <NumericElement
+              name="instrumentOrder"
+              label="Instrument Order"
+              onUserInput={this.setFormData}
+              ref="instrumentOrder"
+              required={false}
+              min="0"
+              max="127"
+              value={this.state.formData.instrumentOrder}
+            />
+            <ButtonElement label="Add entry"/>
+          </FormElement>
+        </div>
+      </div>
+    );
+  }
+
+/** *******************************************************************************
+ *                      ******     Helper methods     *******
+ *********************************************************************************/
+
+  /**
+   * Handle form submission
+   * Check if entry the user is trying to add already exists in the table
+   *
+   * @param {object} e - Form submission event
+   */
+  handleAdd(e) {
+    e.preventDefault();
+
+    let formData = this.state.formData;
+
+    let formObj = new FormData();
+    for (let key in formData) {
+      if (formData[key] !== "") {
+        formObj.append(key, formData[key]);
+      }
+    }
+     // Check if entry the user is trying to add already exists in the table
+     // Pass in callback function to call after check for duplicate
+    this.isDuplicate(formObj, this.giveOptions);
+  }
+
+ /*
+  * Check if entry the user is trying to add already exists in the table
+  */
+  isDuplicate(formObj, callback) {
+    $.ajax({
+      type: 'POST',
+      url: this.props.checkForDuplicate,
+      data: formObj,
+      cache: false,
+      contentType: false,
+      processData: false,
+      success: callback,
+      error: function(err) {
+        console.error(err);
+        swal("Could not check", "", "error");
+      }
+    });
+  }
+
+  /**
+   * Give options depending on duplicate entry:
+   * If there is an inactive duplicate entry, give option to activate it
+   * If there is an active duplicate entry, give no options
+   * If there is no duplicate entry, add entry
+   *
+   * @param {string} duplicateEntry returned by server
+   */
+  giveOptions(duplicateEntry) {
+        // if duplicate entry exists, convert to JSON
+    if (Object.keys(duplicateEntry).length > 0) {
```

Comment:
`duplicateEntry` is a string when you pass it to `Object.keys`
i think some other `isset`-esque check would be more appropriate